ocrd_network/ocrd_network/processing_server.py

```diff
@@ -0,0 +1,301 @@
+                raise HTTPException(
+                    status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                    detail=f"Processor '{processor_name}' not available. It's ocrd_tool is empty or missing"
+                )
+            validator = ParameterValidator(ocrd_tool)
+            report = validator.validate(data.parameters)
+            if not report.is_valid:
+                raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=report.errors)
+
+        # determine path to mets if workspace_id is provided
+        if bool(data.path) == bool(data.workspace_id):
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail="Arguments 'path' and 'workspace_id' are mutually exclusive"
+            )
+        elif data.workspace_id:
+            workspace = await Workspace.get(data.workspace_id)
+            if not workspace:
+                raise HTTPException(
+                    status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                    detail=f"Workspace for id '{data.workspace_id}' not existing"
+                )
+            data.path = workspace.workspace_mets_path
+
+        result_queue_name = None
+        if data.result_queue and data.result_queue.lower() in ["true", "t", "yes", "1"]:
+            result_queue_name = processor_name + '-result'
+
+        job = Job(
+            **data.dict(exclude_unset=True, exclude_none=True),
+            processor_name=processor_name,
+            state=StateEnum.queued,
+            result_queue_name=result_queue_name
+        )
+        await job.insert()
+        processing_message = OcrdProcessingMessage.from_job(job)
+        encoded_processing_message = OcrdProcessingMessage.encode_yml(processing_message)
+        if self.rmq_publisher:
+            self.rmq_publisher.publish_to_queue(processor_name, encoded_processing_message)
+        else:
+            raise Exception('RMQPublisher is not connected')
+        return job.to_job_output()
+
+    async def get_processor_info(self, processor_name) -> Dict:
+        """ Return a processor's ocrd-tool.json
+        """
+        if processor_name not in self.processor_list:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail='Processor not available'
+            )
+        return get_ocrd_tool_json(processor_name)
+
+    async def get_job(self, _processor_name: str, job_id: PydanticObjectId) -> JobOutput:
+        """ Return job-information from the database
+        """
+        job = await Job.get(job_id)
+        if job:
+            return job.to_job_output()
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail='Job not found.'
+        )
+
+    async def list_processors(self) -> str:
+        """ Return a list of all available processors
+        """
+        return json.dumps(self.processor_list)
```

Comment:
Do we really need `json.dumps` here? Simply return `self.processor_list` and FastAPI will automatically serialize it for us, or?

Reply:
I don't see any reason why a `json.dumps` was used there.

> FastAPI will automatically serialize it for us, or?

Yes, since the `processor_list` is basically a python list of strings. `json.dumps` would be required for example if a python dictionary has to be dumped.